src/sstruct_mv/HYPRE_sstruct_grid.c

```diff
@@ -338,8 +338,9 @@ HYPRE_SStructGridSetFEMOrdering( HYPRE_SStructGrid  grid,
       }
    }
 
-   fem_vars    = hypre_TAlloc(HYPRE_Int,  fem_nvars, HYPRE_MEMORY_HOST);
-   fem_offsets = hypre_TAlloc(hypre_Index,  fem_nvars, HYPRE_MEMORY_HOST);
+   fem_vars    = hypre_TReAlloc(hypre_SStructGridFEMPVars(grid, part), HYPRE_Int, fem_nvars, HYPRE_MEMORY_HOST);
```

Comment:
Why is this change needed?

Reply:
There were memory leaks detected. I think ReAlloc can be a fix.